### ISSUE TITLE
Crash on phone number verification screen

### ISSUE THREAD

**Issue by SkewedZeppelin** [signalapp/Signal-Android#13641]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
App immediately crashes upon loading the phone number verification input screen 

### Steps to reproduce
- fresh install
- enter number
- solve captcha
- pin enter screen loads

**Actual result:** immediately crashes
**Expected result:** not crash to allow user to enter the pin

### Device info
**Android version:** latest version of DivestOS based on Android 13, no Google Play Services
**Signal version:**  Signal-Android-website-prod-universal-release-7.12.3.apk

### Link to debug log
```
08-04 19:56:33.204 24857 24894 W Metadata: Failed to find package android.content.pm.PackageManager$NameNotFoundException: com.google.android.gms
08-04 19:56:33.204 24857 24894 E Metadata: Google Play services missing or without correct permission.
08-04 19:56:33.205 24857 24997 W FcmUtil : Failed to get the token.
08-04 19:56:33.205 24857 24997 W FcmUtil : java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: MISSING_INSTANCEID_SERVICE
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.blockingGetToken(FirebaseMessaging.java:632)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.lambda$getToken$4(FirebaseMessaging.java:393)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.$r8$lambda$2qw-XzcEnBhCYM_NFRv12nL9Swg(FirebaseMessaging.java:0)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging$$ExternalSyntheticLambda4.run(R8$$SyntheticClass:0)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.android.gms.common.util.concurrent.zza.run(com.google.android.gms:play-services-basement@@18.2.0:2)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at java.lang.Thread.run(Thread.java:1012)
08-04 19:56:33.205 24857 24997 W FcmUtil : Caused by: java.util.concurrent.ExecutionException: java.io.IOException: MISSING_INSTANCEID_SERVICE
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.android.gms.tasks.Tasks.zza(com.google.android.gms:play-services-tasks@@18.0.2:5)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.android.gms.tasks.Tasks.await(com.google.android.gms:play-services-tasks@@18.0.2:8)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.blockingGetToken(FirebaseMessaging.java:630)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	... 10 more
08-04 19:56:33.205 24857 24997 W FcmUtil : Caused by: java.io.IOException: MISSING_INSTANCEID_SERVICE
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.android.gms.cloudmessaging.Rpc.send(com.google.android.gms:play-services-cloud-messaging@@17.0.0:3)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.GmsRpc.startRpc(GmsRpc.java:227)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.GmsRpc.getToken(GmsRpc.java:180)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.lambda$blockingGetToken$10(FirebaseMessaging.java:617)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.$r8$lambda$lbvegBVb7mfDC14TEL2O2bE8NL0(FirebaseMessaging.java:0)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging$$ExternalSyntheticLambda5.start(R8$$SyntheticClass:0)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.RequestDeduplicator.getOrStartGetTokenRequest(RequestDeduplicator.java:67)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	at com.google.firebase.messaging.FirebaseMessaging.blockingGetToken(FirebaseMessaging.java:613)
08-04 19:56:33.205 24857 24997 W FcmUtil : 	... 10 more
08-04 19:56:33.205 24857 24857 D RegistrationViewModel: FCM token fetched.
08-04 19:56:33.236 24857 24857 D RegistrationViewModel: SMS code request was successful.

08-04 19:56:34.109 24857 24857 D RegistrationViewModel: SMS code request network call completed.
08-04 19:56:34.109 24857 24857 V RegistrationViewModel: handleSessionStateResult()
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: Received MustRetry.
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: org.whispersystems.signalservice.api.push.exceptions.RegistrationRetryException: StatusCode: 429
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket$RegistrationCodeRequestResponseHandler.handle(PushServiceSocket.java:3058)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:2206)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.internal.push.PushServiceSocket.requestVerificationCode(PushServiceSocket.java:418)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode$lambda$3(RegistrationApi.kt:69)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.$r8$lambda$J6AJHpoMZvPa4GtlGHrGnI_pJ6Q(RegistrationApi.kt:0)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi$$ExternalSyntheticLambda4.fetch(R8$$SyntheticClass:0)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.api.NetworkResult$Companion.fromFetch(NetworkResult.kt:38)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.whispersystems.signalservice.api.registration.RegistrationApi.requestSmsVerificationCode(RegistrationApi.kt:68)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at org.thoughtcrime.securesms.registration.data.RegistrationRepository$requestSmsCode$2.invokeSuspend(RegistrationRepository.kt:348)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
08-04 19:56:34.109 24857 24857 I RegistrationViewModel: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)
08-04 19:56:34.115  1366  1432 I am_pss  : [24654,10041,com.android.documentsui,148734976,58466304,83619840,182673408,0,16,18]
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: java.lang.IllegalStateException: Can't access the Fragment View's LifecycleOwner for EnterPhoneNumberFragment{e2f97c5} (1c03d8e5-f3f9-4de0-b0f1-f7917967675a id=0x7f0a08bd tag=6f3402e6-6007-4763-a89a-c2a42ecba070) when getView() is null i.e., before onCreateView() or after onDestroyView()
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at androidx.fragment.app.Fragment.getViewLifecycleOwner(Fragment.java:385)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment.handleErrorResponse(EnterPhoneNumberFragment.kt:305)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment.access$handleErrorResponse(EnterPhoneNumberFragment.kt:72)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment$handlePromptForNoPlayServices$1$1$1.invoke(EnterPhoneNumberFragment.kt:528)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment$handlePromptForNoPlayServices$1$1$1.invoke(EnterPhoneNumberFragment.kt:528)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.handleSessionStateResult(RegistrationViewModel.kt:492)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.requestSmsCodeInternal(RegistrationViewModel.kt:341)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.access$requestSmsCodeInternal(RegistrationViewModel.kt:76)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel$requestSmsCodeInternal$1.invokeSuspend(RegistrationViewModel.kt:0)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at android.os.Handler.handleCallback(Handler.java:942)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at android.os.Handler.dispatchMessage(Handler.java:99)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at android.os.Looper.loopOnce(Looper.java:201)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at android.os.Looper.loop(Looper.java:288)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at java.lang.reflect.Method.invoke(Native Method)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at com.android.internal.os.ExecInit.main(ExecInit.java:48)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
08-04 19:56:34.116 24857 24857 E SignalUncaughtException: 	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@e6f4ad4, Dispatchers.Main.immediate]
08-04 19:56:34.153 24857 24857 I JobManager: Successfully flushed.
08-04 19:56:34.154 24857 24857 E AndroidRuntime: FATAL EXCEPTION: main
08-04 19:56:34.154 24857 24857 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 24857
08-04 19:56:34.154 24857 24857 E AndroidRuntime: java.lang.IllegalStateException: Can't access the Fragment View's LifecycleOwner for EnterPhoneNumberFragment{e2f97c5} (1c03d8e5-f3f9-4de0-b0f1-f7917967675a id=0x7f0a08bd tag=6f3402e6-6007-4763-a89a-c2a42ecba070) when getView() is null i.e., before onCreateView() or after onDestroyView()
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at androidx.fragment.app.Fragment.getViewLifecycleOwner(Fragment.java:385)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment.handleErrorResponse(EnterPhoneNumberFragment.kt:305)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment.access$handleErrorResponse(EnterPhoneNumberFragment.kt:72)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment$handlePromptForNoPlayServices$1$1$1.invoke(EnterPhoneNumberFragment.kt:528)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.phonenumber.EnterPhoneNumberFragment$handlePromptForNoPlayServices$1$1$1.invoke(EnterPhoneNumberFragment.kt:528)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.handleSessionStateResult(RegistrationViewModel.kt:492)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.requestSmsCodeInternal(RegistrationViewModel.kt:341)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel.access$requestSmsCodeInternal(RegistrationViewModel.kt:76)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at org.thoughtcrime.securesms.registration.ui.RegistrationViewModel$requestSmsCodeInternal$1.invokeSuspend(RegistrationViewModel.kt:0)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:942)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:48)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at [[ ↑↑ Original Trace ↑↑ ]].(:0)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at [[ ↓↓ Exception Message ↓↓ ]].(:0)
08-04 19:56:34.154 24857 24857 E AndroidRuntime: 	at Can't access the Fragment View's LifecycleOwner for EnterPhoneNumberFragment{e2f97c5} (********-****-****-****-*********75a id=0x7f0a08bd tag=********-****-****-****-*********070) when getView() is null i.e., before onCreateView() or after onDestroyView().(:0)


08-04 19:56:36.776 25196 25242 I RegistrationUtil: Registration is not yet complete.
08-04 19:56:36.776 25196 25242 I RegistrationUtil: java.lang.Throwable
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at org.thoughtcrime.securesms.registration.RegistrationUtil.maybeMarkRegistrationComplete(RegistrationUtil.java:43)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at org.thoughtcrime.securesms.ApplicationContext.lambda$onCreate$9(ApplicationContext.java:183)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at org.thoughtcrime.securesms.ApplicationContext.$r8$lambda$g4Tr3-ILabGI142ZYkvA84M2oVk(ApplicationContext.java:0)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at org.thoughtcrime.securesms.ApplicationContext$$ExternalSyntheticLambda13.run(R8$$SyntheticClass:0)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at java.lang.Thread.run(Thread.java:1012)
08-04 19:56:36.776 25196 25242 I RegistrationUtil: 	at org.signal.core.util.concurrent.SignalExecutors$NumberedThreadFactory$1.run(SignalExecutors.java:109)

08-04 19:56:37.330 25196 25259 W BaseJob : java.lang.Exception: No release channel recipient.
08-04 19:56:37.330 25196 25259 W BaseJob : 	at org.thoughtcrime.securesms.jobs.StoryOnboardingDownloadJob.onRun(StoryOnboardingDownloadJob.kt:77)
08-04 19:56:37.330 25196 25259 W BaseJob : 	at org.thoughtcrime.securesms.jobs.BaseJob.run(BaseJob.java:31)
08-04 19:56:37.330 25196 25259 W BaseJob : 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:86)
08-04 19:56:37.330 25196 25259 W BaseJob : 	at org.thoughtcrime.securesms.jobmanager.JobRunner.run(JobRunner.java:49)
```

**Comment by nicholas-signal** [comment]
This is fixed in Signal v7.13+ by f87ff5870192df0b1861a99920163e27b516b3aa

**Comment by SkewedZeppelin** [comment]
thank you @nicholas-signal !
